public/app/features/explore/TraceView/components/TracePageHeader/NewTracePageSearchBar.tsx

```diff
@@ -52,11 +65,13 @@ export default memo(function NewTracePageSearchBar(props: TracePageSearchBarProp
     // new query || at end, go to start
     if (prevMatchedIndex === -1 || prevMatchedIndex === spanMatches.length - 1) {
       setFocusedSpanIdForSearch(spanMatches[0]);
+      setCurrentSpanIndex(1);
```

Comment:
`currentSpanIndex` is essentially +1 because it's not array index based, so it does start at 1/32.

`focusedSpanIdForSearch` is used extensively across the trace view and containers. Removing that to suit this one state var in this component would require passing `spanMatches` and `currentSpanIndex` everywhere we want to recalculate `focusedSpanIdForSearch`.

I'd prefer not making a sweeping change right before a release. Perhaps we can merge as is and be mindful of this in the next release.

Reply:
Hey so had to make a few other changes here but overall I'm happy with the approach.